sycl/source/detail/global_handler.hpp

```diff
@@ -126,7 +126,7 @@ class GlobalHandler {
   InstWithLock<std::mutex> MPlatformToDefaultContextCacheMutex;
   InstWithLock<std::mutex> MPlatformMapMutex;
   InstWithLock<std::mutex> MFilterMutex;
-  InstWithLock<std::vector<AdapterPtr>> MAdapters;
+  InstWithLock<std::vector<std::unique_ptr<Adapter>>> MAdapters;
```

Comment:
If we agree that we do not need the `getAdapters` and keep only the `getAdapterRawPtrs`, which returns `std::vector<Adapter *>`, then this line can be changed to the `InstWithLock<std::vector<Adapter*>>` and we can use a custom deleter when constructing the `MAdapters`.

Reply:
And you will not need to change the signature of the `ur::initializeUr()` to return `std::vector` by copy

Reply:
I removed unique_ptr altogether and simplified the code.